### PR TITLE
Todo List Project: CRUD functionality

### PR DESCRIPTION
**Pull request by Profsain**
- Removed all hardcoded items from the tasks array.
- Implemented adding new task functionality.
- Implemented delete task functionality.
- Implemented edit task descriptions functionality.

### CODE REVIEW

File: src/modules/todosclass.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+  }
+
+  // Add new todo
+  addNewTodo() {
+    const newTodos = {
+      id: Date.now(),
+      description: input.value,
+      isCheck: false,
+    };
```
Comment (codepantha) [inline src/modules/todosclass.js]: - [ ] Nice approach with the `id` 👏. However, you still need to add an `index` and set it to the length of the todosArr at the time the new todo item was created. For instance, if a 5th task were added to the list, the index of that task would be equal to 5.

**Please note that deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.** 🎯

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUESTED CHANGES ♻️ 

## Hi @Profsain 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### Highlights ⭐ 
✔️ Descriptive PR
✔️ Meaningful commit messages
✔️ Linter checks
✔️ Good design

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.html [hunk 2]
```diff
@@ -9,9 +9,10 @@
...
-   <form action="#">
-    <input autofocus type="text" name="todo" id="new-todo" placeholder="Add new todo">
-   </form>
+    <form action="#" class="form">
+      <input type="text" name="todo" id="new-todo" placeholder="Add new todo">
+    </form>
+    <p class="feedback"></p>
     <ul class="todo-list">
     </ul>
     <p class="clear-all">Clear all completed</p>
```
Comment (vic778) [inline src/index.html:18]: - [ ] Kindly this `p` should be a button with an `id` so you can use it to clean the array by deleting everything once.

File: src/modules/rendertodolist.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const todoUl = document.querySelector('.todo-list');
+const renderTodoList = ({ id, description, ischeck }) => {
+  const liItems = document.createElement('li');
+  liItems.innerHTML = `
+    <input id="${id}" type="checkbox">
+    <span class="editTodo" id="${id}">${description}</span>
+    <button class="delete-todo" id="${id}" type="button">Delete</button>
+  `;
+
+  todoUl.append(liItems);
```
Comment (vic778) [inline src/modules/rendertodolist.js:10]: - [ ] Kindly this should have an action to make a book completed if the user select the checkbox. add that and let see your project approve :handshake:

**Review by vic778 — CHANGES_REQUESTED** [review]
## Hello @Profsain 

Good job so far! I know you have worked hard on making this project.  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [x] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/rendertodolist.js (hunk 3 above)
Comment (Profsain) [reply src/modules/rendertodolist.js:10]: Hi @vic778 
The project submission is for day 3.
Kindly check the day 3 project requirement for your review and approval. 
Thank you!

**Review by RolandM99 — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hi @Profsain, 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**